src/FirstClaim/FirstClaim.sol

```diff
@@ -0,0 +1,217 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+import {Ownable} from "openzeppelin-contracts/contracts/access/Ownable.sol";
+import {Ownable2Step} from "openzeppelin-contracts/contracts/access/Ownable2Step.sol";
+
+contract FirstClaim is Ownable2Step {
+    ////////////////////////////////////////////////////////////////
+    //                           ERRORS                           //
+    ////////////////////////////////////////////////////////////////
+
+    /// @notice Error that is thrown if the caller is not allowed to call the function
+    error OnlyAllowedCaller();
+
+    /// @notice Error that is thrown if a proposed configuration address is the zero address
+    error ZeroAddress();
+
+    /// @notice Error that is thrown if the grant amount is too large
+    error GrantAmountTooLarge();
+
+    /// @notice Error that is thrown if the owner tries to renounce ownership
+    error CannotRenounceOwnership();
+
+    ////////////////////////////////////////////////////////////////
+    //                           EVENTS                           //
+    ////////////////////////////////////////////////////////////////
+
+    /// @notice Event emitted when a caller is added
+    event CallerAdded(address indexed caller);
+
+    /// @notice Event emitted when a caller is removed
+    event CallerRemoved(address indexed caller);
+
+    /// @notice Event emitted when the AllowanceModule is set
+    event AllowanceModuleSet(address indexed allowanceModule);
+
+    /// @notice Event emitted when the WLD token is set
+    event WldTokenSet(address indexed wldToken);
+
+    /// @notice Event emitted when the Holder is set
+    event HolderSet(address indexed holder);
+
+    /// @notice Event emitted when the RecurringGrantDrop is set
+    event RecurringGrantDropSet(address indexed recurringGrantDrop);
+
+    /// @notice Event emitted when a first claim has been made
+    event FirstClaimClaimed(uint256 grantId, address indexed receiver, uint256 amount);
+
+    ////////////////////////////////////////////////////////////////
+    ///                      CONFIG STORAGE                      ///
+    ////////////////////////////////////////////////////////////////
+
+    /// @notice address of the Safe Allowance Module
+    AllowanceModule public ALLOWANCE_MODULE;
+
+    /// @notice Worldcoin token address
+    address public WLD_TOKEN;
```

Comment:
add immutable

Reply:
it's a configurable field

Reply:
It's irritating that they are in caps: https://docs.soliditylang.org/en/latest/style-guide.html

Reply:
will update